Pavel kirienko patch 1 (#3)
Zubax/dronecode_probe: MANUFACTURING.md

Get the sources from <https://github.com/blacksphere/blackmagic> and build them.
The output will contain the following two files (among others):

* `blackmagic_dfu.bin` - the USB DFU bootloader.
* `blackmagic.bin` - the main firmware.

Prebuilt binaries can be found here: <https://files.zubax.com/products/dronecode_probe>.

The following comands can be used to create a loadable ELF from a flat binary:

    arm-none-eabi-objcopy -I binary -O elf32-little --change-section-address .data=0x08000000 blackmagic_dfu.bin blackmagic_dfu.elf

>     arm-none-eabi-objcopy -I binary -O elf32-little --change-section-address .data=0x08000000 blackmagic.bin blackmagic.elf

Reviewer: Зачем команду дублировать? И так понятно что аргументы нужно подставить какие следует.
```suggestion
```

Author: Не всем понятно. Это же для производственников файл, как я понимаю, не для разработчиков. Для производственников нужно стараться вообще свести к минимуму действия "по умолчанию", должны быть расписаны все операции

Reviewer: Это понятно всем, кто умеет читать; в крайнем случае мы просто отправим им готовый elf.

Author: Ну что плохого если это будет в этом файле? что эта строчка сильно перегружает документ?

Reviewer: Конкретно в этой строчке большой проблемы нет. Просто излишне подробную документацию поддерживать сложнее; со временем детали могут устареть, сделав документацию недействительной. Ошибочная документация куда хуже, чем полностью отсутствующая, поэтому важен баланс. Хорошая документация описывает не последовательность действий, а ключевые концепции, предоставляя читателю досмыслить остальное самостоятельно.

Author: @j3qq4hch ну скажите  Вы что-нибудь. Я точно знаю, что производственники споткнутся при отсутсвии этой строчки и будут нас дергать.

Reviewer: Вообще, если отложить вопрос надобности этой строчки (надобности нет), зачем создавать elf из образа прошивки, ведь для загрузки через бутлоадер нужен бинарник?

Reviewer 2: \*ем попкорн\*

Author: Да, для загрузки чере serial нужен бинарник. Но если прошивать другим DCP через SWD нужен elf.

## Flashing the DFU bootloader

Flash the DFU bootloader image using one of the methods below.

### Using the STM32 serial bootloader

1. Using tweezers, close the jumper `BOOT0` and connect the board to USB
(note that the jumper itself is marked as not to be installed).
The microcontroller will start the embedded bootloader, accessible via the serial port.
Once the board is powered, the tweezers can be removed.

> It is adviced to use the following tool for flashing: <https://github.com/Zubax/zubax_serial_updater>.

Reviewer: Это должно быть в следующем пункте. Ещё тут моя опечатка, правильно `advised`.

2. Using any available STM32 serial bootloader tool, load the DFU bootloader image onto the board at zero offset
(i.e. 0x08000000, which is the address of flash).

3. Disconnect the board from USB.

### Using JTAG/SWD

> 1. Connect the board to USB and SWD pins of the board to any other SWD adapter (e.g., another Dronecode Probe).

Reviewer: ```suggestion
1. Connect the target board to the computer via USB; connect its SWD pins to any other SWD adapter
(e.g., another Dronecode Probe).
```

> SWD pins are marked on the bottom side of the board: SWD, GND, SWC, 5V. Note, that 5V pin is target voltage probe pin. Don't connect it to any power supply.  

Reviewer: ```suggestion
The SWD signals are marked on the bottom side of the board: `SWD`, `GND`, `SWC`, `5V`.
Note that the pin marked `5V` is the target voltage sensing pin (not a power supply input).
```
